tools/ci_build/amd_hipify.py

```diff
@@ -311,6 +303,13 @@ def hipify(src_file_path, dst_file_path):
     s = s.replace("hipdnn", "miopen")
     s = s.replace("HIPDNN_STATUS_SUCCESS", "miopenStatusSuccess")
     s = s.replace("HIPDNN", "MIOPEN")
+    s = s.replace("MIOPEN_BATCHNORM_SPATIAL", "miopenBNSpatial")
```

Comment:
I think it's better we make cudnn_common and miopen_common consistent to have same functions with different impls so that we don't have specific handling in hipify.

Reply:
PR Updated per request. No more `miCompat*` functions.
Replacement of `MIOPEN_BATCHNORM_SPATIAL` and other constants has to be there because HIPDNN was not used in ROCM provider at all.

(Full translation path: `CUDNN_BATCHNORM_SPATIAL` -> (`hipify-perl`) `HIPDNN_BATCHNORM_SPATIAL` -> (`amd_hipify.py` string replacement) `MIOPEN_BATCHNORM_SPATIAL` -> (`amd_hipify.py` string replacement) `miopenBNSpatial`